apps/stepcounter/stepcounter.star

```diff
@@ -0,0 +1,420 @@
+# Permission for you (user) to access the app while it's in test state.
+# Alternately, you may retrieve an Authorization Code via a browser link with redirect to localhost (doesn't really do anything more than the console but was a fun diversion)
+# URL looks like: https://accounts.google.com/o/oauth2/v2/auth?client_id=XXXX.apps.googleusercontent.com&redirect_uri=127.0.0.1&scope=https://www.googleapis.com/auth/userinfo.profile&response_type=code
+# Need to allow the redirect_uri in the credentials console.
+
+# Google API Call and Body to Get a New Access Token.   These expire every 60 minutes.   However, there doesn't seem to be any harm in fetching more often.
+# This script is intended to be invoked once every 30 minutes to refresh step count.
+
+GOOGLE_OAUTH_TOKEN_REFRESH_URL = "https://oauth2.googleapis.com/token"
+GOOGLE_OAUTH_USER_AUTH_URL = "https://accounts.google.com/o/oauth2/v2/auth"
+
+# Random Hash Strings to store Secrets required by Google Oauth (for input to Tidbyt "Secrets" functions)
+# Note - app name is "stepcounter"
+CLIENT_ID_HASH = "AV6+xWcE+SIkQgnPgHzViV78GTRoxpMjlccjdOSUxNRSaBunq5fHKq5xp3sMlKVtYs1V9ZFwBUWg79Pgw+Y3mXoPB5q9AuBVN9bjgND9YpZ9dn3crPs7saefSsj+Mx4K8QUjQgzwLm68+qfWCCtQO419dnPJANjmjXuCrEk02RGw1q3DTRlmaF+Fh+Nf8PRl7wD7Vpfv++8I+WjUlqlhRviULKbMJkyRlZMuBrai"
+CLIENT_SECRET_HASH = "AV6+xWcEb1T8b5kw+ugpxOQ55oRdM9Ox+/PxPSm7V3VTC7NtCrMJXsMA/oozP2Eu8yKUnuDO2jmRB87tsr9ffX1sIkUTdLbftv4swDYku77yz79AJb31q0IRS/gxVkeYuLdgwIt2wqFX6Xrqve2t3wvouaI2WIrpH7U9YzWwc1Iwuv8+6NcKjH0="
+
+# Default (bogus) client ID and Secrets to keep the run time Env happy when running in Deubg Mode
+CLIENT_ID_DEFAULT = "123456"
+CLIENT_SECRET_DEFAULT = "78910"
+
+# Production Code - runs in the Tidbyt production environment.  In Deubg mode with Serve need to hard code valid ID and SECRET here instead (since HASHes above are only valid in production env
+GOOGLE_CLIENT_ID = secret.decrypt(CLIENT_ID_HASH)
+GOOGLE_CLIENT_SECRET = secret.decrypt(CLIENT_SECRET_HASH)
+
+def main(config):
+    # Grab Secrets from paraneters if running in Render mode
+
+    if GOOGLE_CLIENT_ID:
+        client_id = GOOGLE_CLIENT_ID
+    else:
+        client_id = config.get("client_id")
+
+    if GOOGLE_CLIENT_SECRET:
+        client_secret = GOOGLE_CLIENT_SECRET
+    else:
+        client_secret = config.get("client_secret")
+
+    # Google Project Required Info for OAuth
+    # Note to Developers: You need to create your own project in the Google developers console in order to obtain a Client ID and Client Secret.
+
+    # Get the Refresh Token - from Schema if running in Serve/Production mode or via Params when running in Render mode.
+
+    GOOGLEFIT_REFRESH_TOKEN = config.get("auth") or config.get("googlefit_refresh_token")
+
+    # Stepcount cannot be retrieved if a Refresh Token is not present
+    # Render "Unauthorized" in that case
+
+    if not GOOGLEFIT_REFRESH_TOKEN:
+        return render.Root(
+            child = render.WrappedText("Not Authorized (Stepcounter)"),
```

Comment:
Any chance this can show a default view? This is what will show up in the app preview - so a user wouldn't know what the app should look like until they authenticated

Reply:
Done, added new commit on this same branch.   It shows a default screen now.